test/setup/cluster_setup.go

```diff
@@ -279,6 +280,56 @@ func (c *Cluster) StartServer(s *Server) error {
 	return s.start(c.cmdTimeout)
 }
 
+func (c *Cluster) GetServerByID(serverID string) (*Server, int) {
+	if serverID == "" {
+		return nil, -1
+	}
+	for i, srv := range c.Servers {
+		if srv.serverID == serverID {
+			return srv, i
+		}
+	}
+	return nil, -1
+}
+
+func (c *Cluster) AgreedLeader(t *testing.T, activeServers ...int) int {
+	var leaders [3]int
+	var leader int
+
+	for _, srvVal := range activeServers {
+		clusterStatusResEnv, err := c.Servers[srvVal].QueryClusterStatus(t)
+		if err == nil && clusterStatusResEnv != nil {
+			_, leader = c.GetServerByID(clusterStatusResEnv.GetResponse().GetLeader())
+			if leader == -1 {
+				return -1
+			}
+			leaders[srvVal] = leader
+		} else {
+			return -1
+		}
+	}
+
+	for _, srvVal := range activeServers {
+		if leaders[srvVal] != leader {
+			return -1
+		}
+	}
+	return leader
+}
+
+func (c *Cluster) AgreedHeight(t *testing.T, expectedBlockHeight uint64, activeServers ...int) bool {
+	for _, srvVal := range activeServers {
+		blockResEnv, err := c.Servers[srvVal].QueryBlockStatus(t)
+		if blockResEnv != nil {
+			t.Logf("block number = %v", blockResEnv.GetResponse().GetBlockHeader().GetBaseHeader().GetNumber())
+		}
```

Comment:
since we may be polling this every 100ms or so, I don't think it's good to print here.

Reply:
done